include/dgl/runtime/device_api.h

```diff
@@ -57,6 +57,19 @@ class DeviceAPI {
    * \sa DeviceAttrKind
    */
   virtual void GetAttr(DGLContext ctx, DeviceAttrKind kind, DGLRetValue* rv) = 0;
+  /*!
+   * \brief Allocate a data space on device using system mechanisms.
+   * \param ctx The device context to perform operation.
+   * \param nbytes The number of bytes in memory.
+   * \param alignment The alignment of the memory.
+   * \param type_hint The type of elements. Only needed by certain backends such
+   * as OpenGL, as nbytes & alignment are sufficient for most backends.
+   * \return The allocated device pointer.
+   */
+  virtual void* AllocRawDataSpace(DGLContext ctx,
+                               size_t nbytes,
+                               size_t alignment,
+                               DGLType type_hint) = 0;
   /*!
    * \brief Allocate a data space on device.
```

Comment:
Maybe state the difference between this and `AllocRawDataSpace`?  Something like "Allocate a data space on device, potentially with a global workspace caching allocator".